### PR TITLE
[test-framework] Add dynamic port support to policybackend.

### PR DESCRIPTION
**Pull request by ozevren**
Fixes https://github.com/istio/istio/issues/9197

### CODE REVIEW

**Comment by ozevren** [comment]
@ijsnellf

File: pkg/test/framework/components/policybackend/policybackend.go [hunk 1]
```diff
@@ -118,11 +121,19 @@ func (c *localComponent) Init(ctx environment.ComponentContext, deps map[depende
 		return nil, fmt.Errorf("unsupported environment: %q", ctx.Environment().EnvironmentID())
 	}
 
-	port := policy.DefaultPort // TODO: Allow dynamically allocated ports.
+	pm, err := reserveport.NewPortManager()
```
Comment (nmittler) [inline pkg/test/framework/components/policybackend/policybackend.go]: Could you just have the policy backend dynamically assign the port and provide access to it after `Start`?

**Pull request merged**